app/code/community/Bolt/Boltpay/Model/Order.php

```diff
@@ -285,49 +256,321 @@ public function createOrder($reference, $sessionQuoteId = null, $isAjaxRequest =
+     *
+     * event: sales_model_service_quote_submit_before
+     *
+     * @param Varien_Event_Observer $observer Observer event contains an order and (immutable) quote
+     *                                        -  Mage_Sales_Model_Order order
+     *                                        -  Mage_Sales_Model_Quote quote
+     *
+     *                                        The $quote, in turn holds
+     *                                        -  Mage_Sales_Model_Quote parent (ONLY pre-auth; will be empty for admin)
+     *                                        -  object (bolt) transaction (ONLY pre-auth; will be empty for admin)
+     *
+     *
+     * @throws Exception    if an unknown error occurs
+     * @throws Bolt_Boltpay_OrderCreationException if the bottom line price total differs by allowed tolerance
+     *
+     */
+    public function validateBeforeOrderCommit($observer) {
+        /** @var Mage_Sales_Model_Order $order */
+        $order = $observer->getEvent()->getOrder();
+        $payment = $order->getPayment();
+
+        /** @var Mage_Sales_Model_Quote $quote */
+        $immutableQuote = $order->getQuote();
+        $boltTransaction = $immutableQuote->getTransaction();
+
+        if ( (strtolower($payment->getMethod()) !== Bolt_Boltpay_Model_Payment::METHOD_CODE) || empty($boltTransaction) ) {
+            return;
+        }
+
+        /////////////////////////////////////////////////////////////
+        /// When the order is empty, it will not be able to save
+        /// in Magento for an unknown reason.  Here we report the problem
+        /////////////////////////////////////////////////////////////
+        if(empty($order)) {
+            throw new Exception("Order was not able to be saved");
+        }
+        /////////////////////////////////////////////////////////////
+
+        /////////////////////////////////////////////////////////////
+        /// Final sanity check on bottom line price on order.
+        /// If we are failing here, then we've reach an unexpected
+        /// snag that will we
+        /////////////////////////////////////////////////////////////
+        $priceFaultTolerance = $this->boltHelper()->getExtraConfig('priceFaultTolerance');
+
+        $magentoGrandTotal = (int)(round($order->getGrandTotal() * 100));
+        $boltGrandTotal = $boltTransaction->order->cart->total_amount->amount;
+        $totalMismatch = $boltGrandTotal - $magentoGrandTotal;
+
+        if (abs($totalMismatch) > $priceFaultTolerance) {
+            throw new Bolt_Boltpay_OrderCreationException(
+                OCE::E_BOLT_CART_HAS_EXPIRED,
+                OCE::E_BOLT_CART_HAS_EXPIRED_TMPL_GRAND_TOTAL,
+                array($boltGrandTotal, $magentoGrandTotal)
+            );
+        } else if ($totalMismatch) {
+            // Do order total correction if necessary so that the bottom line matches up
+            $order->setTaxAmount($order->getTaxAmount() + ($totalMismatch/100))
```

Comment:
maybe call save just in case

Reply:
aha.  I remember now, this is during the event: `sales_model_service_quote_submit_before`

By definition, these are actions performed precisely before the order is saved.  We do not want to save from this event.  We must respect that other plugins may link into this as well and perform some actions pre-save.  Magento will do the saving here. 

This is the same code section that is described above:
https://github.com/BoltApp/bolt-magento1/blob/548420e59342aa979b2f4667968bf614b46a3302/app/code/community/Bolt/Boltpay/Model/Order.php#L338-L409